src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocLeadingAsteriskAlignCheck.java

```diff
@@ -0,0 +1,202 @@
+                        commentIndex);
+            }
+        }
+        else {
+            DetailNode parent = ast.getParent();
+            while (parent.getType() != JavadocTokenTypes.JAVADOC) {
+                parent = parent.getParent();
+            }
+
+            final int javadocStartTokenColumn = parent.getColumnNumber() - 2;
+            final int commentIndex = getColumnForCommentText(ast.getText());
+
+            checkAsteriskAlignment(
+                    ast.getLineNumber(),
+                    javadocStartTokenColumn,
+                    commentIndex);
+        }
+        isOneLineJavadoc = false;
+        lastLine = "";
+    }
+
+    /**
+     * Create an ArrayList of lines of block comment text to be processed.
+     *
+     * @param ast javadoc end token
+     */
+    private void createLastLine(DetailAST ast) {
+        if (ast.getColumnNo() == 0) {
+            lastLine = ast.getText();
+        }
+        else {
+            final String blockComment = JavadocUtil.getBlockCommentContent(getBlockCommentAst());
+            final List<String> commentTextLineList = Arrays.asList(blockComment.split("\n"));
+
+            if (commentTextLineList.size() <= 1) {
+                isOneLineJavadoc = true;
+            }
+            lastLine = commentTextLineList.get(commentTextLineList.size() - 1) + "*/";
+        }
+    }
+
+    /**
+     * Processes and returns the column number of leading asterisk from javadoc comment text.
+     *
+     * @param line single line from block comment text
+     * @return column number of asterisk
+     */
+    private static int getColumnForCommentText(String line) {
+        int columnNumber = -1;
+        int spaces = 0;
+        final char[] charLine = line.toCharArray();
+
+        int index = 0;
+        while (index < charLine.length) {
+            final char currChar = charLine[index];
+            if (currChar == ' ') {
+                spaces++;
+            }
+            else if (currChar == '*') {
+                columnNumber = spaces;
+                break;
+            }
+            else {
+                index = charLine.length;
+            }
+            index++;
+        }
+        return columnNumber;
+    }
+
+    /**
+     * Checks alignment of asterisks and logs violations.
+     *
+     * @param lineNumber line number of current comment line
+     * @param commentBeginColumn column number of javadoc starting token
+     * @param asteriskColumn column number of current line's leading asterisk
+     */
+    private void checkAsteriskAlignment(int lineNumber,
+                                        int commentBeginColumn,
+                                        int asteriskColumn) {
+
+        if (asteriskColumn != -1) {
+            final int columnDifference = commentBeginColumn - asteriskColumn;
+            if (columnDifference != 0) {
+                log(lineNumber, asteriskColumn, MSG_WRONG_ALIGNMENT, commentBeginColumn + 1);
```

Comment:
How many placeholders does this method call require?

Reply:
1

Reply:
@nrmancuso ping